htdocs/main.inc.php

```diff
@@ -1653,7 +1653,7 @@ function top_menu($head, $title = '', $target = '', $disablejs = 0, $disablehead
 			// Version
 			if (!empty($conf->global->MAIN_SHOWDATABASENAMEINHELPPAGESLINK)) {
 				$langs->load('admin');
-				$appli .= '<br>'.$langs->trans("Database").': '.$db->database_name;
+                $title .= '<br>'.$langs->trans("Database").': '.(!empty($db->database_name)?$db->database_name:$db->db->database_name);
```

Comment:
Do you mean you have some cases where $db->db->database_name and $db->database_name is not ?
When on a page, $db->database_name should always be defined i think.

Reply:
I did not record the commit where the database name was only accessible through $db->db->database_name but as I use always this option I have been annoyed not see it due  to this attribute moving. So I opted to solve it this way.

Reply:
$db should always be Database handler. If not i means you have some bad code that erase it with an object that is a common object class. In such a case, you should fix bad code instead. I suggest to ait you find the case because this change is for a case that should never happen.